open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java

```diff
@@ -1395,13 +1395,25 @@ public SchemaAttributeBuilder  getSchemaAttributeBuilder(SchemaAttribute schemaA
      *
      * @param schemaAttribute bean to interrogate
      * @return guid of the type
+     * @throws InvalidParameterException if the provided schema attribute type cannot be resolved
      */
-    private String getSchemaAttributeTypeGUID(SchemaAttribute   schemaAttribute)
+    private String getSchemaAttributeTypeGUID(SchemaAttribute   schemaAttribute) throws InvalidParameterException
```

Comment:
The typeGUID should be set where the SchemaAttribute object is created.  Isn't there a method like createEmptySchemaAtttibute?  This seems too late.

Reply:
The `SchemaAttribute` object is created by the `clone()` method of the `SchemaAttributeProperties` bean (see line 238 of `AssetOwnerRESTServices`)...  Not sure how I could do this other than extending the bean itself (to include the typedef GUID), or passing through the repository helper to the bean to be able to lookup the typedef GUID (?)

The actual persistence of the `SchemaAttribute` object isn't until after the current point anyway (line 248 of `AssetOwnerRESTServices`), so any exception should break us out before persistence of either the schema type or its attributes...